packages/core/src/config/auth/role-permission-resolver-strategy.ts

```diff
@@ -0,0 +1,24 @@
+import { ID } from '@vendure/common/lib/shared-types';
+
+import { RequestContext } from '../../api';
+import { InjectableStrategy } from '../../common';
+import { User } from '../../entity';
+import { UserChannelPermissions } from '../../service/helpers/utils/get-user-channels-permissions';
+
+/**
+ * @description TODO
+ */
+export interface RolePermissionResolverStrategy extends InjectableStrategy {
+    /**
+     * TODO
+     */
+    persistUserAndTheirRoles(
+        ctx: RequestContext,
+        user: User,
+        /* channelIds: ID[], */ roleIds: ID[],
+    ): Promise<void>;
+    /**
+     * @param user User for which you want to retrieve permissions
+     */
+    resolvePermissions(user: User): Promise<UserChannelPermissions[]>;
```

Comment:
Maybe a bit nitpicky, but more consistent with the rest of the codebase: `saveUserRoles` and `getPermissionsForUser`, instead of resolve/persists?

Reply:
Also, the `channelIds: ID[]` seems pretty crucial for this PoC to work? But the PR is still in draft, right?